### ISSUE TITLE
environment param injection breaks windows authentication

### ISSUE THREAD

**Issue by CodedBeard** [Azure/service-fabric-issues#737]
We are deploying docker-compose containers onto service fabric (6.0.232.9494) in Azure, all of which was working fine, except that environment parameters didn't seem to be coming through to the WebSite even though I could see them in PowerShell. After a bit of reading, I discovered this was due to the IIS site in the container using an apppool other than "DefaultAppPool". After updating the container to use the default apppool instead, the environment params started coming through as expected.

Unfortunately something about how this happens seems to break windows local (not AD) authentication, resulting in a 401.2 error. If i download and run the container locally with passing the environment params via command line, everything works fine. None of the injected settings effect anything to do with authentication (just connection strings for databases) so I'm guessing it has to be some of the additional things that service fabric passes along, or possibly an encoding issue?
I've connected into the service fabric host, and docker exec into the container, I can see that the WebSite configuration is correct, and if i disable Windows authentication and enable anonymous, the site works just fine.

if i run "docker logs" on the service fabric host, i notice there is an error repeated over and over:
```
ERROR ( message:Cannot find requested collection element. )
Applied configuration changes to section "system.applicationHost/applicationPools" for "MACHINE/WEBROOT/APPHOST" at configuration commit path "MACHINE/WEBROOT/APPHOST"
```
This error does not appear when the container is run locally.

**Comment by MikkelHegn** [comment]
Hi @CodedBeard - I don't think the docker logs output is related - you can check with the people over here as this is output from the ServiceMonitor: https://github.com/Microsoft/IIS.ServiceMonitor/.
How are you doing Windows authentication - using domain accounts? If so have you configured the containers for gMSA? Here's a fairly detailed wiki-article around this: https://github.com/artisticcheese/artisticcheesecontainer/wiki/Using-Group-Managed-Service-Account-(GMSA)-to-connect-to-AD-resources

**Comment by CodedBeard** [comment]
Hey @MikkelHegn. No this is local windows auth, so no domain involved. The containers are running an old internal WCF service, which is just being moved off an old server 2003 box to run in containers. As mentioned, if i download the same container (using the full sha so it is definitely the same container) onto server 2016/windows 10 which is not part of the SF cluster and run it there, then the error does not occur. 

I've also tried docker exec into the container, and then confirmed that the local accounts are present, and that the passwords work as expected, so this is certainly an IIS issue, but I've also tried enabling IIS remote management and connected to the container to check the IIS configuration, and it is identical (as you would expect) to the IIS config of the container I downloaded locally. 

The only differences i can find between the container running in SF, and on a local host, is the additional environment variables that SF injects into applicationhost.config, which coupled with the docker logs, is what made me suspect that was the cause.

**Comment by CodedBeard** [comment]
So in the end, this turned out to be something to do with the cluster itself. I never worked out exactly what was causing the error, but after running out of things to try, I decided to release the containers onto the production cluster, where they worked fine. I then built another test cluster (from the same arm template as the previous cluster) and deployed the containers, where again they worked fine with exactly the same settings.
